ext/datadog_profiling_native_extension/heap_recorder.c

```diff
@@ -864,53 +818,24 @@ int heap_frame_cmp(heap_frame *f1, heap_frame *f2) {
   if (line_diff != 0) {
     return line_diff;
   }
-  int cmp = strcmp(f1->name, f2->name);
-  if (cmp != 0) {
-    return cmp;
+  int name_diff = (int) (f1->name - f2->name);
+  if (name_diff != 0) {
+    return name_diff;
   }
-  return strcmp(f1->filename, f2->filename);
+  return (int) (f1->filename - f2->filename);
 }
 
-// TODO: Research potential performance improvements around hashing stuff here
-//       once we have a benchmarking suite.
-//       Example: Each call to st_hash is calling murmur_finish and we may want
-//                to only finish once per structure, not per field?
-//       Example: There may be a more efficient hashing for line that is not the
-//                generic st_hash algorithm?
-
-// WARN: Must be kept in-sync with ::char_slice_hash
-st_index_t string_hash(char *str, st_index_t seed) {
-  return st_hash(str, strlen(str), seed);
-}
-
-// WARN: Must be kept in-sync with ::string_hash
-st_index_t char_slice_hash(ddog_CharSlice char_slice, st_index_t seed) {
-  return st_hash(char_slice.ptr, char_slice.len, seed);
-}
-
-// WARN: Must be kept in-sync with ::ddog_location_hash
 st_index_t heap_frame_hash(heap_frame *frame, st_index_t seed) {
-  st_index_t hash = string_hash(frame->name, seed);
-  hash = string_hash(frame->filename, hash);
+  st_index_t hash = st_hash(&frame->name, sizeof(frame->name), seed);
+  hash = st_hash(&frame->filename, sizeof(frame->filename), hash);
   hash = st_hash(&frame->line, sizeof(frame->line), hash);
```

Comment:
Minor: I guess we could probably hash the whole frame object in one now?

E.g. `st_hash(frame, sizeof(frame), ...)`?